src/extension/background-script/db.ts

```diff
@@ -86,6 +86,21 @@ export class DB extends Dexie {
     return true;
   }
 
+  async clearAllTables() {
+    await this.allowances.clear();
+    await this.payments.clear();
+    await this.blocklist.clear();
+    await this.permissions.clear();
+
+    await browser.storage.local.set({
```

Comment:
There is also browser.storage.local.clear() - would this be better to ensure future entries are also cleared?

Reply:
From what I've figured from the code, there's a sort of replication/mirroring of browser.storage.local and IndexedDB. Also, 'to keep consistency' with the code.

Reply:
I also felt that the notion of "attribute by attribute" clearing would be more verbose - at this point - in contrast with the idea of a whole wipe using browser.storage.local.clear() and the whole wipe in the DB too.